src/linux/frida-helper-backend-glue.c

```diff
@@ -2843,23 +2843,79 @@ frida_find_libc_base (pid_t pid)
 }
 
 #ifdef HAVE_ANDROID
+static gboolean
+frida_find_library_path (pid_t pid, GumAddress base, gchar ** library_path)
+{
+  gboolean result = false;
+  gchar * maps_path;
+  FILE * fp;
+  const guint line_size = 1024 + PATH_MAX;
+  gchar * line, * path;
+
+  *library_path = NULL;
+
+  maps_path = g_strdup_printf ("/proc/%d/maps", pid);
+
+  fp = fopen (maps_path, "r");
+  g_assert (fp != NULL);
+
+  g_free (maps_path);
+
+  line = g_malloc (line_size);
+  path = g_malloc (PATH_MAX);
+
+  while (result == false && fgets (line, line_size, fp) != NULL)
+  {
+    GumAddress start;
+    gint n;
+
+    n = sscanf (line, "%" G_GINT64_MODIFIER "x-%*x %*s %*x %*s %*s %s", &start, path);
+    if (n == 1)
+      continue;
+    g_assert (n == 2);
+
+    if (path[0] == '[')
+      continue;
+
+    if (start == base)
+    {
+      result = true;
+      if (library_path != NULL)
+        *library_path = g_strdup (path);
+    }
+  }
+
+  g_free (path);
+  g_free (line);
+
+  fclose (fp);
+
+  return result;
+}
 
 static GumAddress
 frida_resolve_linker_address (pid_t pid, gpointer func)
 {
   Dl_info info;
-  const gchar * linker_path;
+  gchar * linker_path = NULL;
```

Comment:
```suggestion
  gchar * linker_path;
```

(As it's an out-argument and we trust the implementation to be well-behaved.)

Reply:
On my side this resulted in a compiler warning

Reply:
Oh! So it detects that `frida_find_library_path()` may not always initialize the out-argument?

Reply:
`frida_find_library_path ()` always initialises to NULL

Reply:
I just tried again and I don't get the warning, I am sure I saw it earlier... Maybe it was in the other work in progress branch where I used a slightly different strategy.